modules/pool/invest/lib/useInvest.ts

```diff
@@ -14,13 +14,19 @@ export function useInvest() {
     const { getUserBalanceForToken, userPoolTokenBalances } = usePoolUserTokenBalancesInWallet();
     const { priceForAmount } = useGetTokens();
 
-    const selectedInvestTokens: GqlPoolToken[] = pool.investConfig.options.map((option) =>
-        selectedOptions[`${option.poolTokenIndex}`]
+    const selectedInvestTokens: GqlPoolToken[] = pool.investConfig.options.map((option) => {
+        const firstOptionWithNonZeroBalance = option.tokenOptions.find(
```

Comment:
you only need to do this find if `!selectedOptions[`${option.poolTokenIndex}`]`

Reply:
Refactored this bit.